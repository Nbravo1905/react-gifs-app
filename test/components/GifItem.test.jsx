import { render, screen } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {

  const title = 'One Pice';
  const url = 'https://one-pice.png/';

  test('Prueba match con el snapshot', () => { 
    
    const { container } = render( <GifItem title={ title } url={ url } />);

    expect( container ).toMatchSnapshot();
  });

  test('Debe mostar la imagen con el url y el alt correcto', () => {

    render( <GifItem title={ title } url={ url } />);

    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );

  });

  test('debe mostar el titulo en el componente', () => {

    render( <GifItem title={ title } url={ url } />);
    expect( screen.getByText( title ) ).toBeTruthy();
  })

});