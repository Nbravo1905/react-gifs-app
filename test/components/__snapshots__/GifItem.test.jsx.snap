// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Prueba match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Pice"
      src="https://one-pice.png/"
    />
    <p>
      One Pice
    </p>
  </div>
</div>
`;
